+ Patch Description: Modified the tests to use root file system with xattr support
diff --git a/testcases/kernel/syscalls/llistxattr/llistxattr03.c b/testcases/kernel/syscalls/llistxattr/llistxattr03.c
index 940851f40..ff009e298 100644
--- a/testcases/kernel/syscalls/llistxattr/llistxattr03.c
+++ b/testcases/kernel/syscalls/llistxattr/llistxattr03.c
@@ -28,8 +28,9 @@
 #define SECURITY_KEY   "security.ltptest"
 #define VALUE           "test"
 #define VALUE_SIZE      (sizeof(VALUE) - 1)
+#define MNTPOINT       "mntpoint"

-static const char *filename[] = {"testfile1", "testfile2"};
+static const char *filename[] = {"mntpoint/testfile1", "mntpoint/testfile2"};

 static int check_suitable_buf(const char *name, long size)
 {
@@ -59,6 +60,9 @@ static void verify_llistxattr(unsigned int n)

 static void setup(void)
 {
+       SAFE_MKDIR(MNTPOINT, 0644);
+       SAFE_MOUNT("/dev/vda", MNTPOINT, "ext4", 0, "user_xattr");
+
        SAFE_TOUCH(filename[0], 0644, NULL);

        SAFE_TOUCH(filename[1], 0644, NULL);
@@ -66,12 +70,21 @@ static void setup(void)
        SAFE_LSETXATTR(filename[1], SECURITY_KEY, VALUE, VALUE_SIZE, XATTR_CREATE);
 }

+static void cleanup(void)
+{
+       remove(filename[0]);
+       remove(filename[1]);
+       SAFE_UMOUNT(MNTPOINT);
+       SAFE_RMDIR(MNTPOINT);
+}
+
 static struct tst_test test = {
        .needs_tmpdir = 1,
        .needs_root = 1,
        .test = verify_llistxattr,
        .tcnt = ARRAY_SIZE(filename),
        .setup = setup,
+       .cleanup = cleanup,
 };

 #else /* HAVE_SYS_XATTR_H */

